### DISCUSSION TITLE
XPC connection invalid

### DISCUSSION THREAD

**Discussion by SNewtown** [orchetect/swift-midi#183] [Q&A] [answered]
Hi,

First off, amazing work with this midi wrapper, this will be so much easier to work with than CoreMidi directly. 

To get my bearings, I installed one of your samples (the Bluetooth Midi for iOS section) and while it builds and runs fine, I am getting "XPC connection invalid" when running in simulation and opening the connection config.  It tries to find bluetooth devices but does not list any and it's probably due to the XPC error.   I did add the Plist Bluetooth entry stated in your instructions but still getting the invalid XPC.  Just wondering what else I might be missing or maybe this isn't meant to work in simulation mode? 

Thanks for the help!

Cheers,
Stephane

**Comment by SNewtown** [comment]
Sort of answering my own inquiry... it turns out that it works just fine with a device connected instead of the simulator... so I'm guessing the simulator does not support bluetooth... unless someone has a solution for having bluetooth work within the context of a the simulator...

**Reply by orchetect** [reply, marked as answer]
"XPC connection invalid" or messages similar to XPC connections are usually just log noise and can be ignored. It's not related to what you're doing. Those error messages even come up in a blank app project without Bluetooth or Simulator use.

Re: Bluetooth on an iOS Simulator is not possible. You need to test that on a physical device. (I will add a note to the Bluetooth MIDI example project's README stating that it won't work in a Simulator.)

FWIW, once you have a Bluetooth MIDI session up and running, you interact with the MIDI endpoints it creates just like any other MIDI endpoint. So you could at least bypass the Bluetooth part for testing some of your app in the simulator.
